core/src/main/java/life/qbic/identityaccess/application/user/UserRegistrationService.java

```diff
@@ -173,7 +181,16 @@ public ApplicationResponse requestPasswordReset(String userEmailAddress) {
     if (!user.isActive()) {
       return ApplicationResponse.failureResponse(new UserNotActivatedException("User not active"));
     }
-    DomainEventPublisher.instance().subscribe(new DomainEventSubscriber<PasswordReset>() {
+
+    DomainEventPublisher domainEventPublisher = DomainEventPublisher.instance();
+    while (!domainEventPublisher.clear()) {
+      try {
+        Thread.sleep(1);
+      } catch (InterruptedException ignored) {
+      }
```

Comment:
As I recently learned, swallowing exceptions might lead to a huge overhead when trying to figure out what happened.
(https://www.youtube.com/watch?v=LRwCE7GreSM)
In this case, I would ask that we log the error here or rethrow a RuntimeException and handle the exception on an outer level.
```suggestion
      } catch (InterruptedException ignored) {
        log.error(ignored);
      }
```
```suggestion
      } catch (InterruptedException ignored) {
        throw new RuntimeException("Unexpected interrupt.", ignored);
      }
```